File: backend/src/models/Business.js
const mongoose = require("mongoose");

const businessSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    address: {
      type: String,
      required: true,
    },
    phone: {
      type: String,
      required: true,
    },
    email: {
      type: String,
      required: true,
      unique: true,
    },
    website: {
      type: String,
    },
    owner: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    description: {
      type: String,
    },
    category: {
      type: String,
      enum: [
        "restaurant",
        "retail",
        "service",
        "technology",
        "healthcare",
        "education",
        "real-estate",
        "hospitality",
        "other",
      ],
    },
    askingPrice: {
      type: Number,
      min: 0,
    },
    location: {
      type: String,
    },
    images: [
      {
        url: {
          type: String,
          required: true,
        },
        publicId: {
          type: String,
          required: true,
        },
      },
    ],
    status: {
      type: String,
      enum: ["pending", "approved", "rejected"],
      default: "pending",
    },
    isSold: {
      type: Boolean,
      default: false,
    },
  },
  { timestamps: true }
);

// Add indexes for better query performance
businessSchema.index({ category: 1 });
businessSchema.index({ location: 1 });
businessSchema.index({ askingPrice: 1 });
businessSchema.index({ status: 1 });
businessSchema.index({ owner: 1 });
businessSchema.index({ name: "text", description: "text" }); // Text search index

const Business = mongoose.model("Business", businessSchema);

module.exports = Business;

File: frontend/lib/core/models/business.dart
class Business {
  final String id;
  final String name;
  final String address;
  final String phone;
  final String email;
  final String? website;
  final String? description;
  final String? category;
  final double? askingPrice;
  final String? location;
  final List<BusinessImage> images;
  final String status;
  final bool isSold;
  final DateTime createdAt;
  /// Owner (seller) user id - for contact form sellerRef
  final String? ownerId;

  Business({
    required this.id,
    required this.name,
    required this.address,
    required this.phone,
    required this.email,
    this.website,
    this.description,
    this.category,
    this.askingPrice,
    this.location,
    required this.images,
    required this.status,
    required this.isSold,
    required this.createdAt,
    this.ownerId,
  });

  factory Business.fromJson(Map<String, dynamic> json) {
    final owner = json['owner'];
    final ownerId = owner is Map ? (owner['_id'] as String?) : (owner is String ? owner : null);
    return Business(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'] ?? '',
      website: json['website'],
      description: json['description'],
      category: json['category'],
      askingPrice: (json['askingPrice'] as num?)?.toDouble(),
      location: json['location'],
      images: (json['images'] as List<dynamic>?)
              ?.map((e) => BusinessImage.fromJson(e))
              .toList() ??
          [],
      status: json['status'] ?? 'pending',
      isSold: json['isSold'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      ownerId: ownerId,
    );
  }
}

class BusinessImage {
  final String url;
  final String publicId;

  BusinessImage({
    required this.url,
    required this.publicId,
  });

  factory BusinessImage.fromJson(Map<String, dynamic> json) {
    return BusinessImage(
      url: json['url'] ?? '',
      publicId: json['publicId'] ?? '',
    );
  }
}
